src/layout/TagsInput.js

```diff
@@ -27,13 +28,17 @@ let Tag = observer(({ value, removeTag, tagStyle, onClick }) => (
 ))
 Tag.displayName = 'Tag'
 
-let TagsInput = inject(() => ({
-  state: observable({
+// We're only using withState to preserve the state between renders, since
+// inject doesn't do that for us.
+let TagsInput = withState(
+  'state',
+  'setState',
+  observable({
```

Comment:
Doesn't this need to be `() => observable({})`?

Reply:
That's one option yes, but they also accept a value (https://github.com/acdlite/recompose/blob/master/docs/API.md#withstate)

Reply:
I'm not sure I follow. On the code you linked, the state doesn't get overriden on re-renders since `this.state` is defined outside of `render()`.

Reply:
Not sure how else to explain this other than objects are just references in Javascript. Seems like you're passing the same reference to every instance of the component, which would make them all connected. Could be wrong. This stuff is non-trivial to test locally.

Reply:
You're right. Also it works that way too in pretty much any language I've seen so it was just my brain fart. Fixed now. Also tested.